TaskManager/TaskManager/WeeklyTaskService.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TaskManagerNamespace
+{
+    internal class WeeklyTaskService
+    {
+        private int _counter;
+        internal readonly WeeklyTask[] _tasks;
+
+        public WeeklyTaskService()
+        {
+            _tasks = new WeeklyTask[10];
+        }
+
+        public void HandleAddNewTask()
+        {
+            if (_counter >= 10)
+            {
+                Console.WriteLine("Out of memory. Delete unnecessary tasks, and try again");
+                return;
+            }
+
+            Console.WriteLine("Add new task in format: {},{},{}");
+            var inputData = Console.ReadLine();
+            var task = ParseNewTask(inputData);
+            AddNewTask(task);
+        }
+
+        public void HandleList()
+        {
+            for (int i = 0; i < _counter; i++)
+            {
+                PrintTask(_tasks[i], i);
+            }
+        }
+
+        public void HandleEdit()
+        {
+            Console.WriteLine("Input number to edit:");
+            var inputNumber = Console.ReadLine();
+            var taskNumber = int.Parse(inputNumber);
+
+            Console.WriteLine("Input new task data in format: {},{},{}:");
+            var inputTaskData = Console.ReadLine();
+            var task = ParseNewTask(inputTaskData);
+            _tasks[taskNumber - 1] = task;
+        }
+
+        public void HandleFilterByDate()
+        {
+            Console.WriteLine("Input date:");
+            var inputData = Console.ReadLine();
+            var date = DateTime.Parse(inputData);
+            
+            for (int i = 0; i < _counter; i++)
+            {
+                var task = _tasks[i];
+                if (task.GetDate() >= date)
+                {
+                    PrintTask(task, i);
+                }
+            }
+        }
+
+        private static void PrintTask(WeeklyTask task, int i)
+        {
+            Console.WriteLine(task.ConvertToString(i));
+        }
+
+        public void HandleFilterByPriority()
+        {
+            Console.WriteLine("Input the priority number:");
+            var inputData = Console.ReadLine();
+            var priority = Enum.Parse<Priority>(inputData);
+            if (priority is >= 0 or < (Priority)4)
+            {
+                for (int counterOfPrintTask = 0, i = 0; i < _counter; i++)
+                {
+                    var task = _tasks[i];
+                    if (task.GetPriority() == priority)
+                    {
+                        PrintTask(task, i);
+                        counterOfPrintTask++;
+                    }
+                    if ((_counter - 1 == i) && (counterOfPrintTask == 0))
```

Comment:
instead of another counter you can add a variable `bool hasTasksWithPriority = false;` before the loop. set it to true when print task evaluated. then check after the loop.